### PR TITLE
Fetch Group AAD Id feature

### PR DESCRIPTION
**Pull request by vamagra**
@msftdherron @MattSFT @jialic @aosolis @goelashish7 @jotrick

### CODE REVIEW

File: src/dialogs/examples/teams/FetchTeamsInfoDialog.ts [hunk 1]
```diff
@@ -0,0 +1,49 @@
+import * as builder from "botbuilder";
+import { TriggerActionDialog } from "../../../utils/TriggerActionDialog";
+import { isMessageFromChannel } from "../../../utils/DialogUtils";
+import { DialogIds } from "../../../utils/DialogIds";
+import { DialogMatches } from "../../../utils/DialogMatches";
+import { Strings } from "../../../locale/locale";
+import * as teams from "botbuilder-teams";
+
+export class FetchTeamsInfoDialog extends TriggerActionDialog {
```
[the remark this answers is not in the document]
Comment (vamagra) [reply src/dialogs/examples/teams/FetchTeamsInfoDialog.ts]: Done.

File: src/dialogs/examples/teams/FetchTeamsInfoDialog.ts [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        let msgServiceUrl = msgAddress.serviceUrl;
+
+        // if a message is from a channel, use the team.id to fetch the roster
+        if (!isMessageFromChannel(session.message))
+        {
+            session.send(Strings.aadid_1to1_error);
+        }
+        else
+        {
+            let teamId =  session.message.sourceEvent.team.id;
```
[the remark this answers is not in the document]
Comment (vamagra) [reply src/dialogs/examples/teams/FetchTeamsInfoDialog.ts]: Done.

File: src/dialogs/examples/teams/FetchTeamsInfoDialog.ts [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+            teamsChatConnector.fetchTeamInfo(msgServiceUrl, teamId,
+                (err, result) => {
+                    if (!err) {
+                        session.send(JSON.stringify(result));
```
[the remark this answers is not in the document]
Comment (vamagra) [reply src/dialogs/examples/teams/FetchTeamsInfoDialog.ts]: Done.

File: src/utils/DialogMatches.ts [hunk 4]
```diff
@@ -32,6 +32,7 @@ export const DialogMatches = {
     ChannelDataDialogMatch: /channel data/i,
     DeeplinkDialogMatch: /deeplink/i,
     FetchRosterDialogMatch: /roster/i,
+    FetchTeamsInfoDialogMatch: /aadid/i,
```
[the remark this answers is not in the document]
Comment (vamagra) [reply src/utils/DialogMatches.ts]: Done.

File: src/locale/en/index.json [hunk 5]
```diff
@@ -132,5 +132,6 @@
     "actionable_message_section2_text": "This is an actionable message card. as this section has multiple potential actions in card, action button will be there to expand and hide the controls.",
     "like_message":"You liked a message",
     "remove_like_message":"You removed a 'like' from a message.",
+    "aadid_1to1_error":"AAD Group id doesn't exists for one to one conversation.",
```
[the remark this answers is not in the document]
Comment (vamagra) [reply src/locale/en/index.json]: Done.

File: src/dialogs/examples/teams/FetchTeamsInfoDialog.ts [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                    } else {
+                        session.error(err);
```
[the remark this answers is not in the document]
Comment (vamagra) [reply src/dialogs/examples/teams/FetchTeamsInfoDialog.ts]: user won't see anything here, it just throws the error.

File: src/utils/DialogUtils.ts [hunk 7]
```diff
@@ -185,3 +186,15 @@ export function getTeamId(event: builder.IEvent): string {
     }
     return "";
 }
+
+// Generate the team info data in table format
+export function generateTableForTeamInfo(teamDetails: teams.TeamInfo): string {
```
Comment (aosolis) [inline src/utils/DialogUtils.ts]: We only use this in FetchTeamInfoDialog, let's keep it there.

File: src/dialogs/examples/teams/FetchTeamsInfoDialog.ts (hunk 6 above)
[the remark this answers is not in the document]
Comment (aosolis) [reply src/dialogs/examples/teams/FetchTeamsInfoDialog.ts]: I think you should reply with an error message, like "Error getting team info: " + err.message

File: src/utils/DialogUtils.ts (hunk 7 above)
Comment (vamagra) [reply src/utils/DialogUtils.ts]: Done.

**Pull request merged**